File: extensions/pyRevitDevTools.extension/hooks/view-activated.py
# pylint: skip-file
import os.path as op
from pyrevit import revit, USER_DESKTOP

count = 0
if revit.doc:
    count = len(revit.query.get_all_elements(doc=revit.doc))

with open(op.join(USER_DESKTOP, 'hooks.txt'), 'a') as f:
    f.write('\n'.join([
        'View Activated '.ljust(80, '-'),
        'Cancellable? ' + str(__eventargs__.Cancellable),
        'CurrentActiveView: ' + str(__eventargs__.CurrentActiveView),
        'Document: ' + str(__eventargs__.Document),
        'PreviousActiveView: ' + str(__eventargs__.PreviousActiveView),
        'Status: ' + str(__eventargs__.Status),
        'Element Count: ' + str(count)]) + '\n')

File: extensions/pyRevitDevTools.extension/hooks/view-activated.cs
// executor needs to find the correct function signature and executes that
using System;
using System.IO;

using Autodesk.Revit.DB;
using Autodesk.Revit.UI;


namespace MyEvents {
    public static class MyEventMgr {
        public static void MyEventMgr_UiApp_ViewActivated(object sender, Autodesk.Revit.UI.Events.ViewActivatedEventArgs e) {
            string desktopPath = Environment.GetFolderPath(Environment.SpecialFolder.Desktop);
            File.AppendAllText(
                Path.Combine(desktopPath, "hooks.log"),
                string.Format("[view-activate-csharp] doc:\"{0}\" active_view:\"{1}\" prev_view:\"{2}\" status:\"{3}\"",
                    e.Document.ToString(),
                    e.CurrentActiveView.ToString(),
                    e.PreviousActiveView.ToString(),
                    e.Status.ToString()
                )
            );
        }
    }
}